.github/actions/make/action.yml

```diff
@@ -10,40 +11,74 @@ inputs:
   target-path:
     description: The relative path of the final artifact(s)
     required: true
+  gh-token:
+    required: true
 
 runs:
   using: composite
   steps:
-    - name: Attempt to download the artifact
-      uses: actions/cache/restore@v4
-      with:
-        key: ${{ inputs.key }}
-        path: |
-          ${{ inputs.target-path }}
-          .stamps
-      id: artifact-download
-
-    # This is necessary because the github checkout action doesn't preserve
-    # timestamps on checked out files.
-    - name: Touch downloaded files to update timestamps
-      if: steps.artifact-download.outputs.cache-hit == 'true'
+    - name: Download artifact
       run: |
+        branch="${{ github.head_ref || github.ref_name }}"
+
+        runs_current_branch=$(gh run list --branch "$branch" --limit 10 -w bindings.yml --json databaseId | jq '.[].databaseId')
+        runs_main=$(gh run list --branch "$branch" --limit 10 -w bindings.yml --json databaseId | jq '.[].databaseId')
+
+        # Prioritize current branch over main
+        runs=$(printf "%s\n%s\n" "$runs_current_branch" "$runs_main")
+
+        artifact_name="${{ inputs.key }}"
+
+        echo "found=false" >> $GITHUB_ENV
+
+        # For each run in the sorted list of run ids, try downloading the artifact
+        # with the given name.
+        for run_id in $runs; do
+        if gh run download "$run_id" -n "$artifact_name" >/dev/null 2>&1; then
```

Comment:
This `if` line should be indented to make it use the same indent level as its closing `fi`.